### ISSUE TITLE
Remove SPEC_PASSWORD_EMPTY_PATTERN from BasicAuthentication 

### ISSUE THREAD

**Issue by markfarkas-camunda** [camunda/connectors#1970]
__What should we do?__

We should remove the SPEC_PASSWORD_EMPTY_PATTERN constant from BasicAuthentication class and from easy post connector template. Instead we should just use empty string to indicate that the password was not provided.

__Why should we do it?__

Snyk code scanning finds this code suspicious, and this also adds unnecessary complexity to the code.

**Comment by ev-codes** [comment]
The connectors runtime changes are merged and ready for release!

Adding the new template to the web-modeler [is in review](https://github.com/camunda/web-modeler/pull/8422). Since the required web-modeler reviewer is [busy with code freeze](https://github.com/camunda/web-modeler/pull/8422#issuecomment-1980287272), I'll:
- follow up the reviewer next week
- keep [this issue](https://github.com/camunda/connectors/issues/1970) as "In Review" until the web-modeler PR is merged.

**Comment by ev-codes** [comment]
We discussed today whether we can introduce a new EasyPost template without a password field.

The problem is a new template without a password field will not work with <8.5.0 connectors runtime.

We discussed a few options:
- introducing a new template with the same ID (won't work)
- intorudcing a new template with a new iD and new template type (overly complicated and painful for customers
- add dropdown with "old" and "new" authentication type (overly confusing for customers)
- start deprecation process for "old" template (too painful for customers)

More to come.
